In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn
import numpy as np

Imported libraries together to be more streamlined and clean.

In [7]:
df = pd.read_csv('../Data/Teen_Mental_Health_Dataset.csv')
df.head()

,age,gender,daily_social_media_hours,platform_usage,sleep_hours,screen_time_before_sleep,academic_performance,physical_activity,social_interaction_level,stress_level,anxiety_level,addiction_level,depression_label
0,14,male,7.9,Instagram,7.4,2.9,3.01,1.5,low,2,2,1,0
1,19,female,1.9,TikTok,8.0,2.9,3.22,0.8,high,8,1,10,0
2,17,female,1.3,Instagram,7.6,0.5,3.92,0.0,high,2,4,2,0
3,15,male,7.4,TikTok,6.9,1.6,3.48,0.8,medium,1,7,9,0
4,15,female,4.7,Both,4.9,3.0,2.37,1.4,medium,3,5,2,0


Running basic head, info, value counts, and shape to get information on what to do first for cleanup.

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       1200 non-null   int64  
 1   gender                    1200 non-null   str    
 2   daily_social_media_hours  1200 non-null   float64
 3   platform_usage            1200 non-null   str    
 4   sleep_hours               1200 non-null   float64
 5   screen_time_before_sleep  1200 non-null   float64
 6   academic_performance      1200 non-null   float64
 7   physical_activity         1200 non-null   float64
 8   social_interaction_level  1200 non-null   str    
 9   stress_level              1200 non-null   int64  
 10  anxiety_level             1200 non-null   int64  
 11  addiction_level           1200 non-null   int64  
 12  depression_label          1200 non-null   int64  
dtypes: float64(5), int64(5), str(3)
memory usage: 122.0 KB


1200 rows, 13 columns, no missing values anywhere. This dataset is already in good shape compared to the smartphone addiction dataset. No column needs a null fix.

In [9]:
df.shape

(1200, 13)

In [10]:
df.describe()

,age,daily_social_media_hours,sleep_hours,screen_time_before_sleep,academic_performance,physical_activity,stress_level,anxiety_level,addiction_level,depression_label
count,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000,1200.000000
mean,15.928333,4.536667,6.449417,1.740333,2.990383,1.014500,5.445833,5.636667,5.565000,0.025833
std,2.021947,2.029599,1.442677,0.716660,0.576758,0.582185,2.903290,2.859453,2.830627,0.158704
min,13.000000,1.000000,4.000000,0.500000,2.000000,0.000000,1.000000,1.000000,1.000000,0.000000
25%,14.000000,2.800000,5.200000,1.100000,2.500000,0.500000,3.000000,3.000000,3.000000,0.000000
50%,16.000000,4.500000,6.500000,1.800000,2.990000,1.000000,5.000000,6.000000,6.000000,0.000000
75%,18.000000,6.300000,7.600000,2.400000,3.480000,1.500000,8.000000,8.000000,8.000000,0.000000
max,19.000000,8.000000,9.000000,3.000000,4.000000,2.000000,10.000000,10.000000,10.000000,1.000000


Age runs 13 to 19. Stress, anxiety, and addiction level are all on a 1 to 10 scale and spread evenly across that range, another sign of synthetic data. Academic performance sits between 2.00 and 4.00, reading like a GPA scale. Depression label is 0 or 1.

In [11]:
df['gender'].value_counts()

gender
male      615
female    585
Name: count, dtype: int64

Only two gender values, male and female, split close to even. No 'other' category to handle here, unlike the smartphone dataset.

In [12]:
df.duplicated().sum()

np.int64(0)

Zero duplicate rows.

In [13]:
df['age'].value_counts().sort_index()

age
13    200
14    153
15    180
16    163
17    170
18    172
19    162
Name: count, dtype: int64

Ages spread fairly evenly from 13 to 19 with no outliers.

In [14]:
df.isna().sum()

age                         0
gender                      0
daily_social_media_hours    0
platform_usage              0
sleep_hours                 0
screen_time_before_sleep    0
academic_performance        0
physical_activity           0
social_interaction_level    0
stress_level                0
anxiety_level               0
addiction_level             0
depression_label            0
dtype: int64

Confirms no nulls in any column. Nothing to drop or fill.

In [15]:
df['platform_usage'].value_counts()

platform_usage
Instagram    411
TikTok       398
Both         391
Name: count, dtype: int64

In [16]:
df['social_interaction_level'].value_counts()

social_interaction_level
medium    416
low       415
high      369
Name: count, dtype: int64

Both categorical fields split cleanly into three roughly even groups: Instagram, TikTok, Both for platform, and low, medium, high for social interaction.

In [17]:
df['depression_label'].value_counts()

depression_label
0    1169
1      31
Name: count, dtype: int64

This one is not even. 1169 rows show no depression flag and only 31 show a positive flag, about 2.6 percent of the data. Keeping the column as is since it is a real category count, not a data quality problem, but noting the imbalance now so it does not surprise anyone during analysis.

In [18]:
pd.options.display.float_format = '{:.2f}'.format
df.head()

,age,gender,daily_social_media_hours,platform_usage,sleep_hours,screen_time_before_sleep,academic_performance,physical_activity,social_interaction_level,stress_level,anxiety_level,addiction_level,depression_label
0,14,male,7.90,Instagram,7.40,2.90,3.01,1.50,low,2,2,1,0
1,19,female,1.90,TikTok,8.00,2.90,3.22,0.80,high,8,1,10,0
2,17,female,1.30,Instagram,7.60,0.50,3.92,0.00,high,2,4,2,0
3,15,male,7.40,TikTok,6.90,1.60,3.48,0.80,medium,1,7,9,0
4,15,female,4.70,Both,4.90,3.00,2.37,1.40,medium,3,5,2,0


Set float display to two decimal places to match the formatting used in the smartphone project.

This dataset has no user_id column, unlike both smartphone datasets. Adding one now so it can serve as a primary key in the database step. Using a D1 prefix to keep it distinct from the second dataset, same convention as the smartphone project.

In [19]:
df.insert(0, 'user_id', ['D1_' + str(i).zfill(4) for i in range(1, len(df) + 1)])
df.head()

,user_id,age,gender,daily_social_media_hours,platform_usage,sleep_hours,screen_time_before_sleep,academic_performance,physical_activity,social_interaction_level,stress_level,anxiety_level,addiction_level,depression_label
0,D1_0001,14,male,7.90,Instagram,7.40,2.90,3.01,1.50,low,2,2,1,0
1,D1_0002,19,female,1.90,TikTok,8.00,2.90,3.22,0.80,high,8,1,10,0
2,D1_0003,17,female,1.30,Instagram,7.60,0.50,3.92,0.00,high,2,4,2,0
3,D1_0004,15,male,7.40,TikTok,6.90,1.60,3.48,0.80,medium,1,7,9,0
4,D1_0005,15,female,4.70,Both,4.90,3.00,2.37,1.40,medium,3,5,2,0


In [20]:
df['user_id'].is_unique

True

user_id is unique across all rows.

In [21]:
df.shape

(1200, 14)

In [22]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   1200 non-null   str    
 1   age                       1200 non-null   int64  
 2   gender                    1200 non-null   str    
 3   daily_social_media_hours  1200 non-null   float64
 4   platform_usage            1200 non-null   str    
 5   sleep_hours               1200 non-null   float64
 6   screen_time_before_sleep  1200 non-null   float64
 7   academic_performance      1200 non-null   float64
 8   physical_activity         1200 non-null   float64
 9   social_interaction_level  1200 non-null   str    
 10  stress_level              1200 non-null   int64  
 11  anxiety_level             1200 non-null   int64  
 12  addiction_level           1200 non-null   int64  
 13  depression_label          1200 non-null   int64  
dtypes: float64(5), int6

1200 rows, 14 columns now including user_id. No nulls, no duplicates, all dtypes look right for their fields.

In [23]:
df.to_csv('cleaned_Teen_Mental_Health_Dataset.csv', index=False)

Saved the cleaned data to a new CSV. Original file left unchanged. Second dataset cleanup happens in its own notebook, Data_Cleanup_2nd.ipynb, and both cleaned files come together in Student_Database_SQL.ipynb.